headers/crypto/1.2/psa/crypto-srp.h

```diff
@@ -0,0 +1,182 @@
+/* This file contains reference definitions for implementations of 
+ * SRP-6 in addition to the PSA Certified Crypto API v1.2 PAKE Extension
+ *
+ * These definitions must be embedded in, or included by, psa/crypto.h
+ */
+
+#define PSA_KEY_TYPE_SRP_KEY_PAIR_BASE          ((psa_key_type_t) 0x7700)
+#define PSA_KEY_TYPE_SRP_PUBLIC_KEY_BASE        ((psa_key_type_t) 0x4700)
+#define PSA_KEY_TYPE_SRP_GROUP_MASK             ((psa_key_type_t) 0x00ff)
+
+ /** SRP key pair. Both the client and server key.
```

Comment:
The definition of these SRP-6 keys is technically equivalent to FFDH keys, in terms of key content and calculation of the public key from the private key.

However, there may be a good reason for maintaining a separate key type definition for SRP-6 keys instead of reusing the existing `PSA_KEY_TYPE_DH_KEY_PAIR()`: The derivation of an SRP-6 key is different to the default method for FFDH keys in the Crypto API.
* An FFDH key-pair is derived by repeated sampling of the KDF output to find a value in the required range [1, N-1].
* A SRP-6 key-pair is constructed by zero-padding the fixed-size output of the password-hash. The original SRP hash outputs only 160 bits (for SHA1).

Note to self: this may be worth a rationale comment when defined in the specification.

Reply:
Some descriptions of SRP-6 show the client private key being the output of a simple hash - so we need to ensure the correct construction of an SRP-6 key from an imported data buffer (the hash output), as well as construction via a KDF (such as the original SRP specification double-hash algorithm).

For a general KDF, how does `psa_key_derivation_output_key()` know how much data to extract when deriving the SRP-6 key? The KDF presented in the SRP definition only outputs the hash output size. What about KDFs that do not have limited output? - we can either define the behavior explicitly, or recommend the application extract the required number of KDF bytes, and then import the SRP key?

Reply:
This is a general problem for the derivation of structured keys. The PSA core has to ‘know’ somehow how much data the driver needs to construct the key. The “[psa-driver-interface.md](https://github.com/Mbed-TLS/mbedtls/blob/v3.5.2/docs/proposed/psa-driver-interface.md)” document mentions this under “[Open questions around cooked key derivation](https://github.com/Mbed-TLS/mbedtls/blob/v3.5.2/docs/proposed/psa-driver-interface.md#open-questions-around-cooked-key-derivation)”. The idea of an additional entry point to determine the expected length mentioned there is possibly the best solution.
For trial-and-error type key derivation like FFDH the driver simply returns `PSA_ERROR_INSUFFICIENT_DATA` to signal the core to call it again with the next chunk of data.